### ISSUE TITLE
Optionally fail the build when writing to the build cache fails

### ISSUE THREAD

**Issue by walkerburgin** [gradle/gradle#19171]
### Context
We use Gradle to build a large frontend monorepo. Our CI workflow is structured similar to this: 

![image](https://user-images.githubusercontent.com/1189357/144095425-3454cf71-9e2e-4f6f-8f42-b01fa6d42efc.png)

Each project is built in parallel in its own job and then all of the outputs are assembled into a single artifact that's published by a downstream `publish` job. The Gradle cache is the primary method of communication between these jobs (the `publish` job expects/requires every task to run `FROM CACHE`). For the most part this has worked well for us.  

### Expected Behavior
It would be useful for us if we could configure gradle to fail the build when writing to the build cache fails during a CI build. This would let us attribute failures in our CI workflow to the job that's actually responsible for a downstream failure in the `publish` job. It would also make it easier to retry that job without necessarily having to re-run the entire workflow. 

### Current Behavior
Currently, when writing to the build cache fails, the build succeeds and logs something like this to stdout: 

```
The remote build cache was disabled during the build due to errors.
``` 

As far as I can tell there's no way we can change this behavior.

**Comment by walkerburgin** [comment]
This is still a problem for us!

We're working around it for now by redirecting output to a file and failing the CI workflow if it contains "The remote build cache was disabled during the build due to errors" 🙈 . I'd love for a less brittle way of detecting and acting on these intermittent failures.

**Comment by ov7a** [comment]
Sorry for the late reply.

Thank you for your interest in Gradle!

This issue will be closed as a duplicate of

* https://github.com/gradle/gradle/issues/19338

Please add your use case and 👍 to that issue.
